internal/testcli/golden.go

```diff
@@ -0,0 +1,220 @@
+	if value != nil {
+		if existingMap, ok := value.(*orderedmap.OrderedMap[string, string]); ok {
+			return ctx, existingMap
+		}
+	}
+
+	newMap := orderedmap.NewOrderedMap[string, string]()
+	ctx = context.WithValue(ctx, replacementsMapKey, newMap)
+	return ctx, newMap
+}
+
+func GetReplacementsMap(ctx context.Context) *orderedmap.OrderedMap[string, string] {
+	value := ctx.Value(replacementsMapKey)
+	if value != nil {
+		if existingMap, ok := value.(*orderedmap.OrderedMap[string, string]); ok {
+			return existingMap
+		}
+	}
+	return nil
+}
+
+func setKV(replacements *orderedmap.OrderedMap[string, string], key, value string) {
+	if key == "" || value == "" {
+		return
+	}
+	replacements.Set(key, value)
+}
+
+func PrepareReplacements(t testutil.TestingT, replacements *orderedmap.OrderedMap[string, string], w *databricks.WorkspaceClient) {
+	// in some clouds (gcp) w.Config.Host includes "https://" prefix in others it's really just a host (azure)
+	host := strings.TrimPrefix(strings.TrimPrefix(w.Config.Host, "http://"), "https://")
+	setKV(replacements, host, "$DATABRICKS_HOST")
+	setKV(replacements, w.Config.ClusterID, "$DATABRICKS_CLUSTER_ID")
+	setKV(replacements, w.Config.WarehouseID, "$DATABRICKS_WAREHOUSE_ID")
+	setKV(replacements, w.Config.ServerlessComputeID, "$DATABRICKS_SERVERLESS_COMPUTE_ID")
+	setKV(replacements, w.Config.MetadataServiceURL, "$DATABRICKS_METADATA_SERVICE_URL")
+	setKV(replacements, w.Config.AccountID, "$DATABRICKS_ACCOUNT_ID")
+	setKV(replacements, w.Config.Token, "$DATABRICKS_TOKEN")
+	setKV(replacements, w.Config.Username, "$DATABRICKS_USERNAME")
+	setKV(replacements, w.Config.Password, "$DATABRICKS_PASSWORD")
+	setKV(replacements, w.Config.Profile, "$DATABRICKS_CONFIG_PROFILE")
+	setKV(replacements, w.Config.ConfigFile, "$DATABRICKS_CONFIG_FILE")
+	setKV(replacements, w.Config.GoogleServiceAccount, "$DATABRICKS_GOOGLE_SERVICE_ACCOUNT")
+	setKV(replacements, w.Config.GoogleCredentials, "$GOOGLE_CREDENTIALS")
+	setKV(replacements, w.Config.AzureResourceID, "$DATABRICKS_AZURE_RESOURCE_ID")
+	setKV(replacements, w.Config.AzureClientSecret, "$ARM_CLIENT_SECRET")
+	// setKV(replacements, w.Config.AzureClientID, "$ARM_CLIENT_ID")
+	setKV(replacements, w.Config.AzureClientID, "$USERNAME")
+	setKV(replacements, w.Config.AzureTenantID, "$ARM_TENANT_ID")
+	setKV(replacements, w.Config.ActionsIDTokenRequestURL, "$ACTIONS_ID_TOKEN_REQUEST_URL")
+	setKV(replacements, w.Config.ActionsIDTokenRequestToken, "$ACTIONS_ID_TOKEN_REQUEST_TOKEN")
+	setKV(replacements, w.Config.AzureEnvironment, "$ARM_ENVIRONMENT")
+	setKV(replacements, w.Config.ClientID, "$DATABRICKS_CLIENT_ID")
+	setKV(replacements, w.Config.ClientSecret, "$DATABRICKS_CLIENT_SECRET")
+	setKV(replacements, w.Config.DatabricksCliPath, "$DATABRICKS_CLI_PATH")
+	setKV(replacements, w.Config.AuthType, "$DATABRICKS_AUTH_TYPE")
+}
+
+func PrepareReplacementsUser(t testutil.TestingT, replacements *orderedmap.OrderedMap[string, string], u iam.User) {
```

Comment:
Discussed yesterday: can we replace the ordered map with a slice?

Would rather avoid a new dep for something small.

Reply:
https://github.com/databricks/cli/pull/2025/commits/7f04536a0756510da62868bdab84d00212c4c6d7